vyper/ast/annotation.py

```diff
@@ -22,12 +32,26 @@ def __init__(self, source_code: str, class_types: Optional[ClassTypes] = None):
             self._class_types = {}
 
     def generic_visit(self, node):
-        # Decorate every node in the AST with the original source code. This is
-        # necessary to facilitate error pretty-printing.
+        # Decorate every node with the original source code to allow pretty-printing errors
         node.source_code = self._source_code
         node.node_id = self.counter
+        node.ast_type = node.__class__.__name__
```

Comment:
Why not make this into a `class_name` property of `VyperNode`?

Reply:
It happens in the generic because it has to also happen in `visit_ClassDef` to handle python 3.8's `Constant` types.